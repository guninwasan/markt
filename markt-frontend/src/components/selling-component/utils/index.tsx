export * from "./upload-image";
export * from "./valid-form-details";
