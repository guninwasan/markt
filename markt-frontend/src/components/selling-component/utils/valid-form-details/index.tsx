export * from "./valid-form-details";
