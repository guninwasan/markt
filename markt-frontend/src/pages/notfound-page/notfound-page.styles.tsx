import styled from "styled-components";

const PageContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  text-align: center;
  padding: 2rem;
  background-color: #f9f9f9;
`;

const NotFoundHeading = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: #333;
  margin-bottom: 1rem;
`;

const Message = styled.p`
  font-size: 1.5rem;
  color: #555;
  margin-bottom: 2rem;
`;

const GameContainer = styled.div`
  width: 600px;
  height: 400px;
  max-width: 90%;
  border: none;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
`;

export { PageContainer, NotFoundHeading, Message, GameContainer };
