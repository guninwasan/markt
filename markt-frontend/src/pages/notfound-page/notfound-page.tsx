import React, { useEffect, useRef } from "react";
import { Navbar } from "../../components";
import {
  PageContainer,
  NotFoundHeading,
  Message,
  GameContainer,
} from "./notfound-page.styles";

const NotFoundPage = () => {
  const iframeRef = useRef<HTMLIFrameElement>(null);

  useEffect(() => {
    if (iframeRef.current) {
      iframeRef.current.focus();
    }
  }, []);

  return (
    <>
      <Navbar hideSearch/>
      <PageContainer>
        <NotFoundHeading>Page Not Found</NotFoundHeading>
        <Message>
          Oops! It looks like the page you are looking for does not exist. Enjoy
          this game while we have you here!
        </Message>
        <GameContainer>
          <iframe
            ref={iframeRef}
            src="https://chromedino.com/"
            title="Dino Game"
            width="100%"
            height="100%"
            style={{ border: "none", overflow: "hidden" }}
            scrolling="no"
          ></iframe>
        </GameContainer>
      </PageContainer>
    </>
  );
};

export { NotFoundPage };
