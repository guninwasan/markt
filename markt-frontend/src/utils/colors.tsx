const colors = {
  primary: "#b8d3e0",
  textBlack: "#333",
  black: "#333",
  lightGrey: "#d9d9d9",
  white: "#f9f9f9",
};

export { colors };
