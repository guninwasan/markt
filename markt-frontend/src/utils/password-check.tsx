const passwordCheck = (password: string) => {
  const isAtLeast8Chars = password.length >= 8;
  const hasLowerCase = /[a-z]/.test(password);
  const hasUpperCase = /[A-Z]/.test(password);
  const hasNumber = /\d/.test(password);
  const hasSpecialChar = /[@$!%*?&#-]/.test(password);

  return (
    isAtLeast8Chars &&
    hasLowerCase &&
    hasUpperCase &&
    hasNumber &&
    hasSpecialChar
  );
};

export { passwordCheck };
