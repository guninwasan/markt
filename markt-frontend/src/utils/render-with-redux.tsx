import { render } from "@testing-library/react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux";

export const renderWithRedux = (component: React.ReactNode) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>
  );
};
