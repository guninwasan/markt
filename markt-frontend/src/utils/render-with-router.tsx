import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const renderWithRouter = (component: React.ReactNode) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};
