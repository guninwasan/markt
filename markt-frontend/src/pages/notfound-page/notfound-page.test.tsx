import React from "react";
import { render, screen } from "@testing-library/react";
import { NotFoundPage } from "./notfound-page";
import { renderWithRedux } from "../../utils/render-with-redux";

describe("NotFoundPage", () => {
  it("should render the NotFoundPage component correctly", () => {
    renderWithRedux(<NotFoundPage />);

    expect(screen.getByText("Page Not Found")).toBeInTheDocument();
    expect(
      screen.getByText(
        "Oops! It looks like the page you are looking for does not exist. Enjoy this game while we have you here!"
      )
    ).toBeInTheDocument();
    expect(screen.getByTitle("Dino Game")).toBeInTheDocument();
  });

  it("should focus the iframe element after the component mounts", () => {
    renderWithRedux(<NotFoundPage />);

    const iframeElement = screen.getByTitle("Dino Game");
    expect(document.activeElement).toBe(iframeElement);
  });


});
