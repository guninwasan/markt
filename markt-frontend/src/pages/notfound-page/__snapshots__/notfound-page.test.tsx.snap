// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should match the snapshot 1`] = `
<div>
  <nav
    class="sc-blHHSb isZSSR"
  >
    <a
      aria-current="page"
      class="sc-gtLWhw dEihEB active"
      href="/"
    >
      Markt
    </a>
    <div
      class="sc-eauhAA gTACkZ"
    >
      <div
        class="sc-fFoeYl fxsTkD"
      >
        <input
          class="sc-gQaihK gTtECY"
          placeholder="Search...."
          type="text"
          value=""
        />
        <button
          class="sc-iuUfFv dXDhOe"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </button>
      </div>
    </div>
    <ul
      class="sc-egkSDF ckDOmE"
    >
      <li
        class="sc-fAUdSK dBokxb"
      >
        <a
          class="sc-dntaoT fXFXKj"
          href="/register"
        >
          Login/Register
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="sc-iRVXky uEhcr"
  >
    <h1
      class="sc-ljhlHD iXXOos"
    >
      Page Not Found
    </h1>
    <p
      class="sc-bAEjGW fySqfq"
    >
      Oops! It looks like the page you are looking for does not exist. Enjoy this game while we have you here!
    </p>
    <div
      class="sc-enMaOJ fdyXHS"
    >
      <iframe
        height="100%"
        scrolling="no"
        src="https://chromedino.com/"
        style="overflow: hidden;"
        title="Dino Game"
        width="100%"
      />
    </div>
  </div>
</div>
`;
